src/plugins/ui/header_visibility_ui.ts

```diff
@@ -12,6 +21,66 @@ export class HeaderVisibilityUIPlugin extends UIPlugin {
     "isHeaderHidden",
   ] as const;
 
+  allowDispatch(cmd: Command): CommandResult {
+    switch (cmd.type) {
+      case "REMOVE_COLUMNS_ROWS":
+        if (!this.getters.tryGetSheet(cmd.sheetId)) {
+          return CommandResult.InvalidSheetId;
+        }
+
+        if (this.isAllHeaderIndices(cmd.sheetId, cmd.dimension, cmd.elements)) {
+          this.ui.notifyUI({
```

Comment:
Hi, I am aware that we have updated the API to catch user errors. However, since it was in the master branch, I thought I would update my code there.

Nevertheless, I completely agree with you and will handle this using the example you provided.